cibuildwheel/macos.py

```diff
@@ -83,6 +83,23 @@ def get_python_configurations(
         if any(c.identifier.endswith(a.value) for a in architectures)
     ]
 
+    # filter-out some cross-compilation configs with PyPy:
+    # can't build arm64 on x86_64
+    # rosetta allows to build x86_64 on arm64
+    pypy_architectures = (
+        architectures
+        & {
+            "arm64": {"arm64", "x86_64"},
+            "x86_64": {"x86_64"},
+        }[platform.machine()]
+    )
+    python_configurations = [
+        c
+        for c in python_configurations
+        if (not c.identifier.startswith("pp"))
+        or any(c.identifier.endswith(a.value) for a in pypy_architectures)
+    ]
```

Comment:
maybe this would work and be simpler?

```suggestion
    python_configurations = [
        c
        for c in python_configurations
        if not (c.identifier.startswith("pp") and c.identifier.endswith('arm64') and platform.machine() == 'x86_64')
    ]
```

Reply:
yes, I'm going with a simpler version.